### PR TITLE
Align Eventing backend controller logger with the unified Kyma logger

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Align Eventing backend controller logger with the unified Kyma logger.

> Note: This PR is tested here https://github.com/kyma-project/kyma/pull/11394.

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/11321

### CODE REVIEW

**Review by alexandra-simeonova — APPROVED** [review]

File: components/eventing-controller/cmd/eventing-controller/main.go [hunk 1]
```diff
@@ -2,93 +2,94 @@ package main
...
 	"github.com/kyma-project/kyma/components/eventing-controller/reconciler/backend"
 )
 
 func main() {
-	logger := ctrl.Log.WithName("setup")
-	scheme := runtime.NewScheme()
-	restCfg := ctrl.GetConfigOrDie()
+	setupLogger := ctrl.Log.WithName("setup")
+
+	opts := new(options.Options)
```
Comment (themue) [inline components/eventing-controller/cmd/eventing-controller/main.go:23]: While this is working too I would prefer a today more common `opts := options.New()`.

File: components/eventing-controller/cmd/eventing-controller/main.go [hunk 2]
```diff
@@ -2,93 +2,94 @@ package main
...
+		setupLogger.Error(err, "failed to initialize logger")
+		os.Exit(1)
+	}
+	defer func() {
+		if err := logger.WithContext().Sync(); err != nil {
+			setupLogger.Error(err, "failed to flush logger")
+		}
+	}()
+
+	// set controller core logger
```
Comment (themue) [inline components/eventing-controller/cmd/eventing-controller/main.go:40]: Only an optical change, but we should change even comments to full sentences. Like here

```go
// Set controller core logger.
```

It's latest then needed if a controller has more than one sentence.

File: components/eventing-controller/log/log.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/kyma-project/kyma/common/logging/logger"
+)
+
+type Logger struct {
+	*logger.Logger
+}
+
+// NewLogger returns a new logger with the given format and level
```
Comment (themue) [inline components/eventing-controller/log/log.go:11]: As this is the only returned main type a `log.New()` would be enough.

File: components/eventing-controller/options/options.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	argNameMetricsAddr     = "metrics-addr"
+	argNameReconnectWait   = "reconnect-wait"
+	argNameReconcilePeriod = "reconcile-period"
+
+	// env
+	envNameLogFormat = "APP_LOG_FORMAT"
+	envNameLogLevel  = "APP_LOG_LEVEL"
+)
+
+type Options struct {
```
Comment (themue) [inline components/eventing-controller/options/options.go:23]: I like the idea of an own options package a lot, like the log package. Only would also add an `options.New()`  here. Especially for once possibly needing executed logic when the options instance is created.

Also would add comments for all exported identifiers.

File: components/eventing-controller/log/log.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package log
```
Comment (themue) [inline components/eventing-controller/log/log.go:1]: As calling the main type and all instances logger possibly this package should be called `logger` too.

**Review by themue — COMMENTED** [review]
I do like the idea of those new packages for logger and options. Only have some cosmetically wishes.

**Review by pxsalehi — APPROVED** [review]

**Pull request merged**